packages/rum/test/viewCollection.spec.ts

```diff
@@ -170,7 +172,81 @@ describe('rum track loading time', () => {
     jasmine.clock().tick(AFTER_PAGE_ACTIVITY_END_DELAY)
     jasmine.clock().tick(THROTTLE_VIEW_UPDATE_PERIOD)
 
-    expect(getViewEvent(1).loadingTime).toEqual(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
+    expect(getViewEvent(1).loadingTime).toEqual(msToNs(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY))
+  })
+
+  const FAKE_NAVIGATION_ENTRY = {
+    domComplete: 456,
+    domContentLoadedEventEnd: 345,
+    domInteractive: 234,
+    entryType: 'navigation',
+    loadEventEnd: 567,
+  }
+  const FAKE_NAVIGATION_ENTRY_2 = {
+    domComplete: 2,
+    domContentLoadedEventEnd: 1,
+    domInteractive: 1,
+    entryType: 'navigation',
+    loadEventEnd: 1,
+  }
+
+  it('should use loadEventEnd for initial view when having no activity', () => {
+    expect(getRumEventCount()).toEqual(1)
+    expect(getViewEvent(0).measures).toEqual({
+      errorCount: 0,
+      longTaskCount: 0,
+      resourceCount: 0,
+      userActionCount: 0,
+    })
+
+    lifeCycle.notify(
+      LifeCycleEventType.PERFORMANCE_ENTRY_COLLECTED,
+      FAKE_NAVIGATION_ENTRY as PerformanceNavigationTiming
+    )
+
+    expect(getRumEventCount()).toEqual(1)
+
+    jasmine.clock().tick(THROTTLE_VIEW_UPDATE_PERIOD)
+
+    expect(getRumEventCount()).toEqual(2)
+    expect(getViewEvent(1).loadingTime).toEqual(msToNs(FAKE_NAVIGATION_ENTRY.loadEventEnd))
+  })
+
+  it('should use loadEventEnd for initial view when load event is bigger than computed loading time  ', () => {
+    expect(getRumEventCount()).toEqual(1)
+
+    jasmine.clock().tick(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
+
+    lifeCycle.notify(
+      LifeCycleEventType.PERFORMANCE_ENTRY_COLLECTED,
+      FAKE_NAVIGATION_ENTRY as PerformanceNavigationTiming
+    )
+
+    lifeCycle.notify(LifeCycleEventType.DOM_MUTATED)
+    jasmine.clock().tick(AFTER_PAGE_ACTIVITY_END_DELAY)
+
+    jasmine.clock().tick(THROTTLE_VIEW_UPDATE_PERIOD)
+
+    expect(getRumEventCount()).toEqual(2)
+    expect(FAKE_NAVIGATION_ENTRY.loadEventEnd).toBeGreaterThan(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
+    expect(getViewEvent(1).loadingTime).toEqual(msToNs(FAKE_NAVIGATION_ENTRY.loadEventEnd))
+  })
+
+  it('should not use loadEventEnd for initial view when load event is smaller than computed loading time  ', () => {
+    expect(getRumEventCount()).toEqual(1)
+
+    jasmine.clock().tick(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
+    lifeCycle.notify(
+      LifeCycleEventType.PERFORMANCE_ENTRY_COLLECTED,
+      FAKE_NAVIGATION_ENTRY_2 as PerformanceNavigationTiming
+    )
+    lifeCycle.notify(LifeCycleEventType.DOM_MUTATED)
+    jasmine.clock().tick(AFTER_PAGE_ACTIVITY_END_DELAY)
+    jasmine.clock().tick(THROTTLE_VIEW_UPDATE_PERIOD)
+
+    expect(getRumEventCount()).toEqual(2)
+    expect(FAKE_NAVIGATION_ENTRY_2.loadEventEnd).toBeLessThan(BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY)
```

Comment:
FAKE_NAVIGATION_ENTRY_2.loadEventEnd value does not depend on the behavior of the implementation, no need to check that you have correctly set your tests inputs

Reply:
same comment, I think it gives some context to this test and validate the constant arbitrary values.